graspologic/match/wrappers.py

```diff
@@ -0,0 +1,274 @@
+from collections import namedtuple
+from typing import Any, Optional
+
+import numpy as np
+import pandas as pd
+from beartype import beartype
+from joblib import Parallel, delayed
+from sklearn.utils import check_scalar
+
+from graspologic.match.solver import GraphMatchSolver
+from graspologic.types import Dict, RngType
+
+from .types import (
+    AdjacencyMatrix,
+    InitType,
+    Int,
+    MultilayerAdjacency,
+    PaddingType,
+    Scalar,
+)
+
+MatchResult = namedtuple("MatchResult", ["indices_A", "indices_B", "score", "misc"])
+
+
+@beartype
+def graph_match(
+    A: MultilayerAdjacency,
+    B: MultilayerAdjacency,
+    AB: Optional[MultilayerAdjacency] = None,
+    BA: Optional[MultilayerAdjacency] = None,
+    S: Optional[AdjacencyMatrix] = None,
+    partial_match: Optional[np.ndarray] = None,
+    init: InitType = "barycenter",
+    init_perturbation: Scalar = 0.0,
+    n_init: Int = 1,
+    shuffle_input: bool = True,
+    maximize: bool = True,
+    padding: PaddingType = "naive",
+    n_jobs: Optional[Int] = None,
+    max_iter: Int = 30,
+    tol: Scalar = 0.01,
+    verbose: Int = 0,
+    rng: Optional[RngType] = None,
+    use_numba: bool = False,
```

Comment:
Maybe just remove this for now - from a public api perspective I can see the appeal of only making these changes like, once, but nobody can even use it even if they set it to True, so I would argue this would be a bit of a breaking change either way (as in, if I set it to True, and used the NotImplementedError for some sort of control flow, because I'm a maniac, then use silently adding numba support would result in different behavior between versions.  So we'd still have to do the appropriate semver incrementation).

Thus, I'd say just remove it for now, and feel free to link to a github work item saying "add numba support to graph matching" in the docs or something

Reply:
yeah, to clarify i would not be adding this unless it worked, this is one of the in-progress items for this PR (sorry for the confusion there). i think it might take like 2 lines of work to make numba happy, but if not, then yes ill take out all numba references and just make an issue

Reply:
opted to remove numba for now